compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -224,13 +224,15 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 terminator.kind
             ),
 
-            InlineAsm { template, ref operands, options, destination, .. } => {
+            InlineAsm { template, ref operands, options, ref targets, .. } => {
                 M::eval_inline_asm(self, template, operands, options)?;
                 if options.contains(InlineAsmOptions::NORETURN) {
                     throw_ub_custom!(fluent::const_eval_noreturn_asm_returned);
                 }
                 self.go_to_block(
-                    destination
+                    targets
+                        .get(0)
+                        .copied()
```

Comment:
This should instead just not do `go_to_block` at all and leave it to `eval_inline_asm` to do the jump.

Please also add a doc comment [here](https://github.com/rust-lang/rust/blob/4d93590d59ca7d71ac401ce600fd0fe458a6375e/compiler/rustc_const_eval/src/interpret/machine.rs#L376) saying that the function is expected to handle the jumping to the next block.